tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java

```diff
@@ -284,7 +296,7 @@ public static void main(String[] args) {
 
         String consumerGroup = parsedArgs.getString("consumerGroup");
 
-        final KafkaProducer<String, String> producer = createProducer(parsedArgs);
+        AtomicReference<KafkaProducer<String, String>> producer = new AtomicReference<>(createProducer(parsedArgs));
```

Comment:
volatiles can't be local. Arguably this could get refactored a little bit if we don't want to call get everywhere.